client/components/command-pallette/README.md

```diff
@@ -0,0 +1,33 @@
+# Command Palette
+
+The Command Palette is a React component used to display a text input used to search and filter a list of commands. It can be accessed in the Calypso using the shortcut `cmd+k`.
+
+The command palette is inspired by [Core](https://github.com/WordPress/gutenberg/blob/trunk/packages/commands/README.md) implementation used in the Gutenberg editor.
+
+## Add a command
+
+In order to add a new command to the palette, you can add them in `useCommandsArrayWpcom` or create a new hook and include it to `useCommandPallette` hooks.
+
+## Command properties
+
+Every command has a set of properties that can be used to customize its behavior. The following properties are available:
+
+- `name`: Type `string`, used as identifier for the command. It's ex
+- `label`: Type `string`, used as the label for the command. Visible in the command palette.
+- `subLabel`?: Type `string`, used as a second line for the command. Visible in the command palette.
+- `searchLabel`?: Type `string`, used to match the user search. It's not visible for the user. We recommend to use `_x()` to add context to the translators.
+- `callback`: Type `function`, used to execute the command when the user clicks on it or press enter on that command.
+- `context`?: Type `string[]`, a list of paths used to display the command by default.
+- `icon`?: Type `JSX.Element`, used to display an icon for the command. It's visible in the command palette.
+- `image`?: Type `JSX.Element`, used to display an image for the command, instead of an icon. It's visible in the command palette.
+- `siteFunctions`?: Type `SiteFunctions`, used for nested commands that need to execute a function when a site is selected in the command palette as a second step.
+
+## Usage
```

Comment:
What do you think about showing how to define example command, too?

Reply:
I added a command example:

https://github.com/Automattic/wp-calypso/pull/84591/commits/6e4db1601bac2cfa95076542ce33600f14ee13dc

Reply:
Thanks! Do you think we should keep this example simple, or could we use it to show how to use other fields e.g. `searchLabel`?

```
searchLabel: _x( 'add / create a new site', 'used as a keywords when user searches for commands' ),
```

Reply:
Seems like example should be first up, and embedding bits aren't something folks need to worry about so it can be lower.